applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,242 +60,273 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
     }
 
     /**
-     * Sets key value. Label value for the configurationSetting is optional, if not specified or label=%00 it implies null label.
-     * @param configurationSetting key and value to set
-     * @return ConfigurationSetting that was created or updated
+     * Gets the default pipeline policies.
+     * TODO (conniey): This is going to change when we move to a builder pattern.
      */
-    public Mono<RestResponse<ConfigurationSetting>> setKeyValue(ConfigurationSetting configurationSetting) {
-        return setKeyValue(configurationSetting, null);
+    public static List<HttpPipelinePolicy> getDefaultPolicies(String connectionString) {
+        final ApplicationConfigCredentials credentials = new ApplicationConfigCredentials(connectionString);
+        // Closest to API goes first, closest to wire goes last.
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+        policies.add(new UserAgentPolicy(String.format("Azure-SDK-For-Java/%s (%s)", SDK_NAME, SDK_VERSION)));
+        policies.add(new RequestIdPolicy());
+        policies.add(new RetryPolicy());
+        policies.add(new ConfigurationCredentialsPolicy(credentials));
+
+        return policies;
     }
 
     /**
-     * Sets key value. Label value for the configurationSetting is optional, if not specified or label=%00 it implies null label.
-     * @param configurationSetting key and value to set
+     * Adds a configuration value in the service if that key and label does not exist.
+     * <p>
+     * The label value for the ConfigurationSetting is optional. If not specified, the
+     * {@link ConfigurationSetting#NULL_LABEL} is used.
+     *
+     * @param configurationSetting The key, value, and label to set.
      * @return ConfigurationSetting that was created or updated
+     * @throws com.microsoft.azure.v3.CloudException when a ConfigurationSetting with the same key and label exists.
      */
-    public Mono<RestResponse<ConfigurationSetting>> setKeyValue(ConfigurationSetting configurationSetting, ETagFilter filter) {
+    public Mono<RestResponse<ConfigurationSetting>> addKeyValue(ConfigurationSetting configurationSetting) {
```

Comment:
Let's discuss - I want to be careful whenever we introduce new terminology (in this case the 'add')/